### DISCUSSION TITLE
eslint `consistent-return` error after migration to SvelteKit v2 (redirect, error)

### DISCUSSION THREAD

**Discussion by DePasqualeOrg** [sveltejs/kit#11467] [Q&A] [answered]
Now that `redirect` and `error` are implicitly thrown, I get an eslint error for the `consistent-return` rule in functions that use them, like this for example:

```typescript
export const actions = {
  deleteAccount: async ({ locals }) => {
    const user = await authenticate(locals);
    if (user) {
      await prisma.user.delete({
        where: { email: user.email },
      });
      return {};
    }
    redirect(303, '/');
  },
} satisfies Actions;

```

`Expected to return a value at the end of async method 'deleteAccount'.eslint[consistent-return](https://eslint.org/docs/latest/rules/consistent-return)`

How can this be resolved?

**Comment by Magnus-Ignitic** [comment, marked as answer]
Putting a `return` in front of `redirect` works, but feels like a hacky workaround. You might argue it makes code more readable though, clearly showing that redirect exits the function:

```js
export const actions = {
  deleteAccount: async ({ locals }) => {
    const user = await authenticate(locals);
    if (user) {
      await prisma.user.delete({
        where: { email: user.email },
      });
      return {};
    }
    return redirect(303, '/');
  },
} satisfies Actions;
```

The only other option would be to disable the rule for that function:

```js
// eslint-disable-next-line consistent-return
```

**Reply by teemingc** [reply]
You can also do `throw redirect(303, '/')` which is arguably more accurate but also a hacky workaround since it's the same as `redirect(303, '/')`.
